lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -1244,13 +1275,22 @@ Error OpenCLFunction::execute(ExecutionContext *context) {
       if (src == dest) {
         continue;
       }
-      size_t destOff = runtimeBundle_.getValueOffset(dest);
-      size_t srcOff = runtimeBundle_.getValueOffset(src);
+
+      cl_mem destBuf = clBindings->getBuffer(dest);
+      cl_mem srcBuf = clBindings->getBuffer(src);
       size_t sizeInBytes = dest->getSizeInBytes();
       cl_event event{nullptr};
-      cl_int err = clEnqueueCopyBuffer(commands, deviceBuffer, deviceBuffer,
-                                       srcOff, destOff, sizeInBytes, 0, nullptr,
-                                       kernelProfiling_ ? &event : nullptr);
+      cl_int err =
+          clEnqueueCopyBuffer(commands, srcBuf, destBuf, 0, 0, sizeInBytes, 0,
+                              nullptr, kernelProfiling_ ? &event : nullptr);
+      llvm::outs() << "COPY\n";
```

Comment:
Please remove debug prints.

Reply:
just temporary I'm trying to printf debug the POCL build. will fix before landing

Reply:
Did you notice POCL_DEBUG=1 which might be useful?